cibuild.cmd

```diff
@@ -18,9 +18,30 @@ if /I "%1" == "/release" set BuildConfiguration=Release&&shift&& goto :ParseArgu
 if /I "%1" == "/test32" set Test64=false&&shift&& goto :ParseArguments
 if /I "%1" == "/test64" set Test64=true&&shift&& goto :ParseArguments
 if /I "%1" == "/testDeterminism" set TestDeterminism=true&&shift&& goto :ParseArguments
+
+REM /buildTimeLimit is the time limit, measured in minutes, for the Jenkins job that runs
+REM the build. The Jenkins script netci.groovy passes the time limit to this script.
+if /I "%1" == "/buildTimeLimit" set BuildTimeLimit=%2&&shift&&shift&&goto :ParseArguments
+
 call :Usage && exit /b 1
 :DoneParsing
 
+REM This script takes the presence of the /buildTimeLimit option as an indication that it
+REM should run the tests under the control of the ProcessWatchdog, which, if the tests
+REM exceed the time limit, will take a screenshot, obtain memory dumps from the test
+REM process and all its descendants, and shut those processes down.
+REM
+REM Developers building from the command line will presumably not pass /buildTimeLimit,
+REM and so the tests will not run under the ProcessWatchdog.
+if not "%BuildTimeLimit%" == "" (
+    set __CurrentDate=%date%
+    set __CurrentTime=%time%
```

Comment:
Just to emphasize that it's a local variable, but I see the script doesn't use that convention elsewhere, so I'll change it.